pkg/queue/cluster_queue_impl.go

```diff
@@ -221,6 +222,25 @@ func (c *clusterQueueBase) DumpInadmissible() (sets.Set[string], bool) {
 	return elements, true
 }
 
+func (c *clusterQueueBase) Snapshot() ([]*workload.Info, bool) {
```

Comment:
Let's use `kueue.Workload`, rather than `workload.Info`, because workload is all we need for the API.

Reply:
for sort operation we use `queueOrdering` function that does it take workload.Info objects. Of course, we can iterate again and put this in kueue.Workload, but why?

Reply:
I was thinking about returning workloads to keep the API cleaner, as we don't need the other bits of information, such as `TotalRequests` or `ClusterQueue`. It also means GC can cleanup the info objects later. 

However, I now see the tradeoff, that using info allows to save one pass, so feel free to keep as is.